lib/ElementHandle.js

```diff
@@ -149,6 +156,30 @@ class ElementHandle extends JSHandle {
     return {x, y, width, height};
   }
 
+  /**
+   * @return {!Promise<?object>}
+   */
+  async boxModel() {
+    const result = await this._getBoxModel();
+
+    if (!result)
+      return null;
+
+    return Object.keys(result.model).reduce((prev, key) => {
```

Comment:
I'm sure this works, but I'm having hard times following the logic.
I'd rather have something more straight-forward here that outlines the shape of the returned object.

This will make it clear what we return (and also improves code grep'pability!)

```js
return {
  content: fromProtocolQuad(result.content),
  padding: fromProtocolQuad(result.padding),
  // ...
}

function fromProtocolQuad(protocolQuad) {
  // ...
}
```

Reply:
done